src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/VsUIService`1.cs

```diff
@@ -15,16 +15,14 @@ internal class VsUIService<T> : IVsUIService<T>
     {
         private readonly Lazy<T> _value;
         private readonly IProjectThreadingService _threadingService;
-        private readonly IServiceProvider _serviceProvider;
 
         [ImportingConstructor]
         public VsUIService([Import(typeof(SVsServiceProvider))]IServiceProvider serviceProvider, IProjectThreadingService threadingService)
         {
             Requires.NotNull(serviceProvider, nameof(serviceProvider));
             Requires.NotNull(threadingService, nameof(threadingService));
 
-            _value = new Lazy<T>(GetService);
-            _serviceProvider = serviceProvider;
+            _value = new Lazy<T>(() => (T)serviceProvider.GetService(ServiceType));
             _threadingService = threadingService;
```

Comment:
ThreadHelper complicates unit tests, and causes downstream issues everytime we side-by-side Shell due to how it gets initialized. I'm avoiding using it deliberately. In side-by-side situations the first component that touches it initializes it (instead of the shell), which might not be on the UI thread.